src/NServiceBus.Core/Pipeline/MainPipelineExecutor.cs

```diff
@@ -15,7 +14,9 @@ class MainPipelineExecutor(
     INotificationSubscriptions<ReceivePipelineCompleted> receivePipelineNotification,
     IPipeline<ITransportReceiveContext> receivePipeline,
     IActivityFactory activityFactory,
-    MessagingMetricsMeters messagingMetricsMeters)
+    PipelineMetrics pipelineMetrics,
+    string queueNameBase,
+    string endpointDiscriminator)
```

Comment:
These two parameters sound off. They are needed down below to initialize default tags in the IncomingPipelineMetricTags, but still 🤷